src/main/scala/cromwell/engine/workflow/WorkflowOptions.scala

```diff
@@ -0,0 +1,89 @@
+package cromwell.engine.workflow
+
+import com.typesafe.config.ConfigFactory
+import cromwell.util.{EncryptedBytes, SecretKey, Aes256Cbc}
+import scala.collection.JavaConverters._
+import spray.json._
+import scala.util.{Try, Success, Failure}
+
+object WorkflowOptions {
+  private lazy val WorkflowOptionsConf = ConfigFactory.load.getConfig("workflow-options")
+  private lazy val EncryptedFields: Seq[String] =
+    WorkflowOptionsConf.getList("encrypted-fields").unwrapped().asScala.toSeq.asInstanceOf[Seq[String]]
+  private lazy val EncryptionKey: String =
+    WorkflowOptionsConf.getString("base64-encryption-key")
+
+  def encryptField(value: JsString): Try[JsObject] = {
+    Aes256Cbc.encrypt(value.value.getBytes("utf-8"), SecretKey(EncryptionKey)) match {
+      case Success(encryptedValue) => Success(JsObject(Map(
+        "iv" -> JsString(encryptedValue.base64Iv),
+        "ciphertext" -> JsString(encryptedValue.base64CipherText)
+      )))
+      case Failure(ex) => Failure(ex)
+    }
+  }
+
+  def decryptField(obj: JsObject): Try[String] = {
+    (obj.fields.get("iv"), obj.fields.get("ciphertext")) match {
+      case (Some(iv: JsString), Some(ciphertext: JsString)) =>
+        Aes256Cbc.decrypt(EncryptedBytes(ciphertext.value, iv.value), SecretKey(WorkflowOptions.EncryptionKey)) match {
+          case Success(decryptedBytes) => Success(new String(decryptedBytes, "utf-8"))
+          case Failure(ex) => Failure(ex)
+        }
+      case _ => Failure(new Throwable(s"JsObject must have 'iv' and 'ciphertext' fields to decrypt: $obj"))
+    }
+  }
+
+  def isEncryptedField(jsValue: JsValue): Boolean = jsValue match {
+    case obj: JsObject if obj.fields.keys.exists(_ == "iv") && obj.fields.keys.exists(_ == "ciphertext") => true
+    case _ => false
+  }
+
+  def fromJsonObject(jsObject: JsObject): Try[WorkflowOptions] = {
+    val encrypted = jsObject.fields map {
+      case (k, v: JsString) if EncryptedFields.contains(k) => k -> encryptField(v)
+      case (k, v: JsString) => k -> Success(v)
+      case (k, v) if isEncryptedField(v) => k -> Success(v)
+      case (k, v) => k -> Failure(new Throwable(s"Unsupported key/value pair in WorkflowOptions: $k -> $v"))
+    }
+
+    encrypted.values collect { case f: Failure[_] => f } match {
```

Comment:
How do you make the types work out?

`encrypted.values collect { case f: Failure[_] => f }` returns a `Iterable[Try[JsValue] with Failure]`
`encrypted.values.filter(_.isFailure)` only returns an `Iterable[Try[JsValue]]` so I'd have to do a `.asInstanceOf` to use them as Failures?

Is there a better way?